src/librustc/infer/error_reporting/mod.rs

```diff
@@ -1140,8 +1141,20 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                         &"type",
                         expected,
                         found,
-                        &format!(" ({})", values.expected.sort_string(self.tcx)),
-                        &format!(" ({})", values.found.sort_string(self.tcx)),
+                        &format!(" ({})",
+                             if let ty::Opaque(def_id, _) = values.expected.sty {
```

Comment:
The only difference between this and the one below is `values.expected` vs. `values.found`. It should be de-duplicated.

Reply:
Thanks - I'd looked at adding a function but didn't quite seem worth it. A closure on the other had... looks much neater now.